app/models/form526_submission.rb

```diff
@@ -630,27 +612,47 @@ def cleanup
     EVSS::DisabilityCompensationForm::SubmitForm526Cleanup.perform_async(id)
   end
 
-  def find_user_account_with_icn(records, record_type)
-    records.pluck(:user_account_id).uniq.each do |user_account_id|
-      user_account = UserAccount.find(user_account_id)
-      next if user_account&.icn.blank?
-
-      Rails.logger.info("ICN not found on submission #{id}, " \
-                        "using ICN for user account #{user_account_id} instead (based on #{record_type})")
-      return user_account
+  def get_icn_from_mpi
+    edipi_response_profile = edipi_mpi_profile_query(auth_headers['va_eauth_dodedipnid'])
+    if edipi_response_profile&.icn.present?
+      OpenStruct.new(icn: edipi_response_profile.icn)
+    else
+      Rails.logger.info("Form526Submission::account - unable to look up MPI profile with EDIPI for user #{user_uuid}")
+      attributes_response_profile = attributes_mpi_profile_query(auth_headers)
+      if attributes_response_profile&.icn.present?
+        OpenStruct.new(icn: attributes_response_profile.icn)
+      else
+        Rails.logger.info("Form526Submission::account - no ICN present for user #{user_uuid}")
+        OpenStruct.new(icn: nil)
+      end
     end
   end
 
-  def get_past_submissions
-    Form526Submission.where(user_uuid:).where.not(user_account_id:)
+  def edipi_mpi_profile_query(edipi)
+    return unless edipi
+
+    edipi_response = mpi_service.find_profile_by_edipi(edipi:)
+    edipi_response.profile if edipi_response.ok? && edipi_response.profile.icn.present?
+  end
+
+  def attributes_mpi_profile_query(auth_headers)
+    required_attributes = %w[va_eauth_firstName va_eauth_lastName va_eauth_birthdate va_eauth_pnid]
+    return unless required_attributes.all? { |attr| auth_headers[attr].present? }
```

Comment:
You said in your comment before that this could potentially return even if ssn or one of these is missing, but here you are returning right away if one is missing, is that intentional?

Reply:
It turns out that I was incorrect on this, the `lib/mpi/messages/find_profile_by_attributes.rb` validates for all 4 of those attributes before querying MPI so they're all required.